dynamic_analysis_pkg/dynamic_analysis_pkg/__init__.py

```diff
@@ -0,0 +1,167 @@
+            return
+        self.retrieve_sim_info(token, sim_id)
+        self.get_sim_results()
+
+    def retrieve_sim_info(self, token, sim_id=None):
+        tr = unicore_client.Transport(token)
+        job_url = join(self.job_base, sim_id)
+        job = unicore_client.Job(tr, job_url)
+        print('Fetching results of [{}]. Please wait ...'.format(job.properties['name']))
+        storage = job.working_dir
+        self.files_list = storage.listdir()
+
+    def get_sim_results(self):
+        self.download_blueconfig()
+        self.download_report()
+        self.download_out_dat()
+        print('Result were saved at: {}'.format(self.local_dir))
+
+    def define_paths(self, sim_id, working_directory, new_circuit_path):
+        self.local_dir = join(working_directory, sim_id)
+        if not exists(self.local_dir):
+            makedirs(self.local_dir)
+
+        if not isdir(new_circuit_path):
+            print('[ERROR] Circuit path does not exists')
+        else:
+            self.circuit_dir = new_circuit_path
+
+        self.ORIGIN_DIR_PATH = join(self.ORIGIN_DIR_PATH, sim_id)
+
+    def download_file_to_storage(self, file_name):
+        # function to download the file and add it to kernel file system
+        file_output_path = join(self.local_dir, file_name)
+        if exists(file_output_path):
+            print('- [{}] file already exists. Skipping download.'.format(file_name))
+            return
+
+        print('- Fetching [{}] ...'.format(file_name))
+        x = self.files_list[file_name]
+
+        if file_name == 'BlueConfig':
+            file_content = x.raw().read()
+            bc_str = file_content.decode('utf-8')
+            writable_content = bc_str.replace(self.ORIGIN_CIRCUIT_PATH, self.circuit_dir)
+            writable_content = writable_content.replace(self.ORIGIN_DIR_PATH, self.local_dir)
+
+            with open(file_output_path, 'w') as fd:
+                fd.write(writable_content)
+
+        else:
+            x.download(str(file_name)) # moves to home always
+            move(file_name, file_output_path)
+
+        print('- [{}] downloaded'.format(file_name))
+
+    def download_blueconfig(self, file_name='BlueConfig'):
+        self.download_file_to_storage(file_name)
+
+    def download_report(self):
+        reports = [x for x in self.files_list if '.bbp' in x]
+        if not reports:
+            print('No reports were found')
+            return
+        for report in reports:
+            self.download_file_to_storage(report)
+
+    def download_out_dat(self):
+        self.download_file_to_storage('out.dat')
+
+
+class FetchMultipleResults:
+
+    values = []
+
+    def __init__(self, token, sim_list_ids, base_working_directory=None):
+
+        total_sims = len(sim_list_ids)
+        for idx, sim in enumerate(sim_list_ids):
+            print('({}/{})'.format(idx + 1, total_sims))
+            self.values.append(Results(token, sim, base_working_directory))
```

Comment:
You can have a progress bar with:
```python
from tqdm import tqdm
```

```suggestion
    for sim in tqdm(sim_list_ids):
            self.values.append(Results(token, sim, base_working_directory))
```